src/shogun/machine/BaggingMachine.cpp

```diff
@@ -226,12 +230,17 @@ bool CBaggingMachine::train_machine(CFeatures* data)
 
 		#pragma omp critical
 		{
-		// get out of bag indexes
-		CDynamicArray<index_t>* oob = get_oob_indices(idx);
-		m_oob_indices->push_back(oob);
+			// get out of bag indexes
+			CDynamicArray<index_t>* oob = get_oob_indices(idx);
+			m_oob_indices->push_back(oob);
 
-		// add trained machine to bag array
-		m_bags->push_back(c);
+			// add trained machine to bag array
+			m_bags->push_back(c);
+
+			// observe some variables
+			auto oob_error = this->get_oob_error();
+			this->observe<CMachine*>(i, "trained_machine", "Trained machine for this bag", c);
```

Comment:
this reminds me of a question I had, two actually:

* How are the machines (or general: objects) stored in the observations? Because if just the pointer is stored and the objects are modified further downstream of the algorithm, then this breaks stuff. Don't they need to be cloned?
* For the printing (the logger observer), could we pls use the visitor pattern of Any that is used in `CSGObject::to_string()`? See that method for how to do that